### PR TITLE
Update vcpkg baseline to the October 2022 tag.

### PR DESCRIPTION
**Pull request by ahsonkhan**

### CODE REVIEW

File: sdk/samples/iot/paho_iot_hub_c2d_sample.c [hunk 1]
```diff
@@ -12,7 +12,7 @@
 // warning C4201: nonstandard extension used: nameless struct/union
 #pragma warning(disable : 4201)
 #endif
-#include <paho-mqtt/MQTTClient.h>
+#include <MQTTClient.h>
```
Comment (ahsonkhan) [inline sdk/samples/iot/paho_iot_hub_c2d_sample.c:15]: This is a follow-up to https://github.com/Azure/azure-sdk-for-c/pull/2377#issuecomment-1299323996

I am assuming this is the only mitigation needed within the samples when upgrading the vcpkg baseline and the paho-mqtt dependency.

File: vcpkg.json [hunk 2]
```diff
@@ -12,11 +12,5 @@
     {
       "name": "cmocka"
     }
-  ],
-  "overrides": [
-    {
-      "name": "zlib",
-      "version": "1.2.12#2"
-    }
```
Comment (ahsonkhan) [inline vcpkg.json:20]: Reverting https://github.com/Azure/azure-sdk-for-c/pull/2377

With the latest baseline (if it builds successfully), the overrides shouldn't be necessary.

File: cmake-modules/AzureVcpkg.cmake [hunk 3]
```diff
@@ -21,7 +21,7 @@ macro(az_vcpkg_integrate)
       if(NOT DEFINED ENV{AZURE_SDK_DISABLE_AUTO_VCPKG})
         # GET VCPKG FROM SOURCE
         #  User can set env var AZURE_SDK_VCPKG_COMMIT to pick the VCPKG commit to fetch
-        set(VCPKG_COMMIT_STRING f0aa678b7471497f1adedcc99f40e1599ad22f69) # default SDK tested commit
+        set(VCPKG_COMMIT_STRING 94ce0dab56f4d8ba6bd631ba59ed682b02d45c46) # default SDK tested commit
```
Comment (ahsonkhan) [inline cmake-modules/AzureVcpkg.cmake:24]: This commit is the  same as the newest released tag: `tags/2022.10.19`

**Review by antkmsft — APPROVED** [review]
I'd get ok from the IoT team, and also let's update the C++ repo to the same tag/commit SHA.

**Comment by ahsonkhan** [comment]
Looks like all the linux legs are failing because the vcpkg version installed is different than the ones we need. I am wondering if it is a vcpkg caching issue, or possibly the vcpkg version pre-installed on the linux machines are out-dated and need to be fixed.

```text
-- Running vcpkg install
Error: while checking out baseline from commit '94ce0dab56f4d8ba6bd631ba59ed682b02d45c46' at subpath 'versions/baseline.json':
error: git failed with exit code: (128).
fatal: path 'versions/baseline.json' exists on disk, but not in '94ce0dab56f4d8ba6bd631ba59ed682b02d45c46'

This may be fixed by updating vcpkg to the latest master via `git pull` or fetching commits via `git fetch`.

You can use the current commit as a baseline, which is:
    "builtin-baseline": "ce99c947b64ea3c0e274e18cc3035ca727681089"
-- Running vcpkg install - failed
CMake Error at /usr/local/share/vcpkg/scripts/buildsystems/vcpkg.cmake:868 (message):
  vcpkg install failed.  See logs for more information:
  /mnt/vss/_work/1/s/build/vcpkg-manifest-install.log
Call Stack (most recent call first):
  /usr/local/share/cmake-3.24/Modules/CMakeDetermineSystem.cmake:124 (include)
  CMakeLists.txt:40 (project)
```

cc @danieljurek, @weshaggard, @BillyONeal

**Comment by danewalton** [comment]
> 

As long as the E2E tests pass and it compiles looks okay to me.

**Review by danewalton — APPROVED** [review]

**Comment by weshaggard** [comment]
@ahsonkhan I'm not an expert but based on the issue https://github.com/microsoft/vcpkg/issues/23748 it looks like it might have to do something with the builtin-baseline sha.

**Comment by BillyONeal** [comment]
>Error: while checking out baseline from commit '94ce0dab56f4d8ba6bd631ba59ed682b02d45c46' at subpath 'versions/baseline.json':
error: git failed with exit code: (128).
fatal: path 'versions/baseline.json' exists on disk, but not in '94ce0dab56f4d8ba6bd631ba59ed682b02d45c46'

This probably means you're missing a fetch? https://github.com/microsoft/vcpkg/commit/94ce0dab56f4d8ba6bd631ba59ed682b02d45c46 exists.

**Comment by ahsonkhan** [comment]
@BillyONeal, I can try adding a `vcpkg fetch` step in our CI pipelines/machines, but I am wondering why this would **only** fail on linux. Does linux machines in AzDo come with an existing clone of vcpkg (or vcpkg pre-bootstrapped/installed)?

**Comment by BillyONeal** [comment]
> Otherwise, why would a fetch be required on a clean clone of a repo, to checkout an older commit sha?

I'm not sure, is this actually a clone? Unfortunately I don't know about how you invoke us and it isn't obvious from this change. Doing a `which vcpkg` may help check that we're getting the right copy?

I do observe that the commit you're explicitly trying to checkout is older than the baseline you're trying to use, so there may be versions referenced by that baseline that are unreachable by the version database in use?

**Comment by ahsonkhan** [comment]
> I do observe that the commit you're explicitly trying to checkout is older than the baseline you're trying to use

Really?
The commit sha I am checking out should be identical to the baseline. I am using the released tag:
94ce0dab56f4d8ba6bd631ba59ed682b02d45c46 (tags/2022.10.19)
https://github.com/microsoft/vcpkg/releases/tag/2022.10.19

**Comment by ahsonkhan** [comment]
Fallout issues discovered from this work:
https://github.com/Azure/azure-sdk-for-c/issues/2403
https://github.com/Azure/azure-sdk-for-c/issues/2404

File: eng/scripts/vcpkg.ps1 [hunk 4]
```diff
@@ -16,9 +16,11 @@ Param (
 $initialDirectory = Get-Location
 
 try {
+    Write-Host "Cloning into: $VcpkgPath"
```
Comment (vhvb1989) [inline eng/scripts/vcpkg.ps1:19]: better to just remove this file. It's death code

File: eng/vcpkg-commit.txt [hunk 5]
```diff
@@ -1 +1 @@
-tags/2020.06
+tags/2022.10.19
```
Comment (vhvb1989) [inline eng/vcpkg-commit.txt:1]: You can also remove this file

**Review by vhvb1989 — APPROVED** [review]

**Comment by ahsonkhan** [comment]
Based on recommendation from Billy, we decided to avoid using `VCPKG_ROOT` or a "system-level" vcpkg repo, and instead use a repo-specific/local copy of vcpkg. This way we don't rely on the pre-installed version that happens to come with the build machines.

**Pull request merged**